### ISSUE TITLE
C++ forward computation error

### ISSUE THREAD

**Issue by Aditi138** [clab/dynet#1295]
I am writing a simple loss function `for (int32_t n = 0; n <= a->neg; n++) {
                if (n == 0) {
                  Expression context_emb;
                  context_emb = lookup(cg, context, unsigned(line[w + c]));
                  Expression dot_prod = dot_product(input_emb, context_emb);
                  errs[n] = log_sigmoid(dot_prod);
                  cout << context_emb.dim() << endl;
                  cout << dot_prod.dim() << endl;
                  cout << errs[n].value() << endll;
                } else {
                  int32_t negative_sample;
                  negative_sample = utils.getNegative(line[w]);
                  Expression negative_emb = lookup(cg, context, unsigned(negative_sample));
                  Expression dot_prod = dot_product(input_emb, negative_emb);
                  errs[n] = log_sigmoid(-1 * dot_prod);
                  cout << negative_emb.dim() << endl;
                  cout << dot_prod.dim() << endl;
                  cout << errs[n].value() << endl;
                }
              }
               Expression loss = -sum(errs);
loss_val += as_scalar(cg.forward(loss));`
The print statements are displaying valid outputs and dimenstion, but when it fails at the forward step and gives BAD ACCESS exception and goes into this file "dynet/exec.cc" at this line `     DYNET_ARG_CHECK(xs[ai]->device == node->device ||
            node->supports_multidevice(),`

Any advice?
